lib/engine/game/g_18_oe/step/buy_sell_par_shares.rb

```diff
@@ -16,23 +26,69 @@ def actions(entity)
             actions << 'par' if can_ipo_any?(entity) || can_float_minor?(entity)
             actions << 'buy_company' if !purchasable_companies(entity).empty? || !buyable_bank_owned_companies(entity).empty?
             actions << 'sell_shares' if can_sell_any?(entity)
-
-            actions << 'pass' if !can_float_minor?(entity) && !bought? && !actions.empty?
+            actions << 'convert' unless @can_convert.empty?
+            actions << 'pass' if !can_float_minor?(entity) && !actions.empty?
             actions
           end
 
+          def corporation_actions(entity)
+            return [] unless @can_convert.include?(entity)
+
+            %w[convert pass]
+          end
+
           def can_buy_any_from_ipo?(entity)
             return unless @game.corporations.all?(&:ipoed)
 
             super
           end
 
+          def can_buy?(entity, bundle)
+            action = @round.current_actions.find { |a| a.is_a?(Engine::Action::Convert) }
+            return false if action && action&.entity != bundle.corporation
```

Comment:
Could there be multiple convert actions in `@round.current_actions`? If there are, then you will just be checking the first one to see whether it matches `bundle.corporation`.
Would this be better?
```
return false if @round.current_actions.any? { |a| a.is_a?(Engine::Action::Convert) && a.entity == bundle.corporation }
```

Reply:
There can only be one convert action done per turn, so there should only be 0 or 1 in current_actions